sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExternalSource.scala

```diff
@@ -121,6 +130,10 @@ object ExternalSource {
   }
 
   def getScans: Map[Class[_ <: Scan], ScanRule[_ <: Scan]] = {
+    if (showErrorMessage) {
```

Comment:
I think we should consolidate all this logic in hasSparkAvroJar. I'll make a PR to your PR  with a suggestion